cachi2/core/errors.py

```diff
@@ -1,14 +1,18 @@
 from enum import Enum
 
 
+class Cachi2Error(Exception):
```

Comment:
Why do we benefit from a generic Cachi2 error? Just to separate our exceptions from others?

Reply:
I'm just afraid we'll end up with something like in Cachito, where every single error was caught and rethrown as a CachitoError, ultimately making debugging much harder. :face_exhaling: 

If you look at an error like NetworkError, then I am not sure if we should put this under a "expected to happen" type of error. It feels more like Cachi2 errors are "known" errors.

Also, client x server errors don't make sense anymore, and are not being properly used even in the code that was copied pasted (looking at you [FileAccessError](https://github.com/containerbuildsystem/cachi2/blob/132c000044c5ecf1c4307a0b1dab933c93ed1a3f/cachi2/core/package_managers/gomod.py#L67))

I'd say we should file a ticket and design this error handling better, and ASAP.

Reply:
The thing that hides all unexpected errors behind "An unknown error occurred" is this https://github.com/containerbuildsystem/cachito/blob/7662e45d43f74e96a1b745112688c0b4ed7b9e62/cachito/workers/tasks/general.py#L120

If we don't do that, then we won't end up in that situation :smile: 

But yeah, I will file that ticket

Reply:
What I meant was more in the lines of:

```
try: 
  something()
except SpecifcException:
  raise CachitoError()
```

But yeah, I'm not against having a CachitoError, let's just clarify it's purpose.

Reply:
Changed the docstring